docs/life-of-a-packet.md

```diff
@@ -0,0 +1,17 @@
+# Life of a Packet
+
+```mermaid
+flowchart LR
+  A(Apps) <--> S(Socket)
+  subgraph OS
+    S <--> R(Routing Table)
+  end
+  R <-- Outline OFF --> IF(Network Interface)
+  IF <--> INET(((Internet)))
+  R <-- Outline ON --> TUN(Tun Device)
+  TUN <--> T2S(Run2socks)
+  subgraph Outline Client
+    T2S <--> SC(SSClient)
```

Comment:
The DNS fallback to TCP is not covered by this graph, but I think it's ok for now.